x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/MlSingleNodeTestCase.java

```diff
@@ -96,20 +96,7 @@ protected NamedXContentRegistry xContentRegistry() {
     public void tearDown() throws Exception {
         try {
             logger.trace("[{}#{}]: ML-specific after test cleanup", getTestClass().getSimpleName(), getTestName());
-            String[] nonAnnotationMlIndices;
-            boolean mlAnnotationsIndexExists;
-            do {
-                String[] mlIndices = client().admin().indices().prepareGetIndex().addIndices(".ml-*").get().indices();
-                nonAnnotationMlIndices = Arrays.stream(mlIndices).filter(name -> name.startsWith(".ml-annotations") == false)
-                    .toArray(String[]::new);
-                mlAnnotationsIndexExists = mlIndices.length > nonAnnotationMlIndices.length;
-            } while (nonAnnotationMlIndices.length > 0 && mlAnnotationsIndexExists == false);
-            if (nonAnnotationMlIndices.length > 0) {
-                // Delete the ML indices apart from the annotations index.  The annotations index will be deleted by the
-                // base class cleanup.  We want to delete all the others first so that the annotations index doesn't get
-                // automatically recreated.
-                assertAcked(client().admin().indices().prepareDelete(nonAnnotationMlIndices).get());
-            }
+            client().execute(ResetFeatureStateAction.INSTANCE, new ResetFeatureStateRequest()).actionGet();
```

Comment:
I think you should also delete the comment on lines 91-94, as this cleanup is now doing more than what that comment says.  It will mean that in the future it will be possible to use this base class for tests that might leave jobs running when they complete.